crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -179,51 +183,52 @@ impl ModuleSubscriptionActor {
     }
 
     fn remove_subscriber(&mut self, client_id: ClientActorId) {
-        self.subscriptions.retain_mut(|sub| {
+        let mut subscriptions = self.subscriptions.write_arc();
+        subscriptions.retain_mut(|sub| {
             sub.remove_subscriber(client_id);
             !sub.subscribers().is_empty()
         })
     }
 
-    async fn _broadcast_commit_event(&mut self, mut event: ModuleEvent, tx: &mut Tx) -> Result<(), DBError> {
-        let futures = FuturesUnordered::new();
+    async fn broadcast_commit_event(&mut self, mut event: ModuleEvent) -> Result<(), DBError> {
```

Comment:
This function really shouldn't be async. The reason being is that subscription evaluations need to have a serial equivalent ordering to the original transactions.

If txn `A` inserts a row, and then txn `B` deletes that same row, it would be incorrect to evaluate the effects of `B` before the effects of `A`.

Note this only applies to incremental evaluation `eval_incr`. It does not apply to the initial subscribe call `eval` because of the absence of deletes.

Reply:
I got you, This would require some more work I guess. Thanks for pointing.

Reply:
@Shubham8287 just reiterating the solution that we discussed earlier:

If each reducer holds an exclusive lock on the `ModuleSubscriptionManager`, we can evaluate all of the subscription queries for **single** transaction concurrently, and at the same time preserve the ordering across transactions.

That is, reducers are still run sequentially and do not complete until all of the required subscription queries have been evaluated. However within said reducer, queries can be run in parallel.

Reply:
Thanks for summarizing.
I have done changed the we intented to achive parrallelism, so that now instead of trying to run `broadcast_commit_event` parallelly. The method will spawn threads to evaluate each Subsription `QuerySet`s
and keeping subscription actor itself single threaded.